src/lib/logical_query_plan/projection_node.cpp

```diff
@@ -36,17 +36,20 @@ std::string ProjectionNode::description() const {
   return desc.str();
 }
 
-std::shared_ptr<AbstractLQPNode> ProjectionNode::_deep_copy_impl() const {
-  std::vector<std::shared_ptr<Expression>> column_expressions;
-  column_expressions.reserve(_column_expressions.size());
-  for (const auto& expression : _column_expressions) {
-    column_expressions.emplace_back(expression->deep_copy());
-  }
+std::shared_ptr<AbstractLQPNode> ProjectionNode::_deep_copy_impl(
+    const std::shared_ptr<AbstractLQPNode>& left_child, const std::shared_ptr<AbstractLQPNode>& right_child) const {
+  Assert(this->left_child() && left_child, "Can't deep copy without child to adjust ColumnOrigins");
```

Comment:
I don't understand this error message. This might be because I don't understand the function in general.